qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -0,0 +1,250 @@
+                 quantum_instance: Optional[Union[QuantumInstance, BaseBackend, Backend]] = None
+                 ) -> None:
+        """Initializes the Circuit Quantum Neural Network.
+
+        Args:
+            circuit: The (parametrized) quantum circuit that generates the samples of this network.
+            input_params: The parameters of the circuit corresponding to the input.
+            weight_params: The parameters of the circuit corresponding to the trainable weights.
+            interpret: Determines the output format, possible choices are:
+                * 'tuple' (default): a tuple of binary values, e.g. (0, 1, 0, 1, 0)
+                * 'str': a bitstring of type str, e.g. '01010'
+                * 'int': an integer corresponding to the bitstring, e.g. 10
+                * a custom callable that takes a sample of type 'tuple' and maps it to some other
+                output, output should be hashable for sparse representation of probabilities
+                and probability gradients.
+            dense: Whether to return a dense (array with 'output_shape') or sparse (dict)
+                probabilities. Dense probabilities require "interpret == 'int'" where the integer
+                will be the index in the array of probabilities.
+                TODO: what about "return_samples"??? (cf. base class)
+                TODO: update return types to handle dictionaries and arrays
+            output_shape: Gives the output_shape in case of a custom interpret callable. If this is
+                None, the output_shape is set to 1.
+            gradient: The gradient converter to be used for the probability gradients.
+            quantum_instance: The quantum instance to evaluate the circuits.
+
+        Raises:
+            QiskitMachineLearningError: if an incorrect value for `interpret` or `output_shape`
+                is passed.
+        """
+
+        # TODO: currently cannot handle statevector simulator, at least throw exception
+
+        # copy circuit and add measurements in case non are given
+        self._circuit = circuit.copy()
+        if quantum_instance.is_statevector:
+            if len(self._circuit.clbits) > 0:
+                self._circuit.remove_final_measurements()
+        elif len(self._circuit.clbits) == 0:
+            self._circuit.measure_all()
+
+        self._input_params = list(input_params or [])
+        self._weight_params = list(weight_params or [])
+        self._interpret = interpret
+        self._dense = dense
+        self._gradient = gradient
+
+        if isinstance(quantum_instance, (BaseBackend, Backend)):
+            quantum_instance = QuantumInstance(quantum_instance)
+        self._quantum_instance = quantum_instance
+
+        # TODO this should not be necessary... but currently prop grads fail otherwise
+        from qiskit import Aer
+        self._sampler = CircuitSampler(Aer.get_backend('statevector_simulator'), param_qobj=False)
```

Comment:
Ok, hopefully we don't need that. If there was any need to do something we should use BasicAer since Aer is still really an optional install and not part of Qiskit core (i.e. Terra)

Reply:
Dependency on Aer is removed.